src/main/java/org/broadinstitute/hellbender/tools/spark/sv/CollectInsertSizeMetricsSpark.java

```diff
@@ -138,24 +140,28 @@ public int getValue(){
     private static final String R_SCRIPT = "insertSizeHistogram.R";
 
     @Override
-    protected void runTool(final JavaSparkContext ctx) {
+    public boolean requiresReads(){return true;}
+
+    @Override
+    public void runTool(final JavaSparkContext ctx) {
 
         final JavaRDD<GATKRead> filteredReads = getReads();
 
+        final SAMFileHeader readsHeader = getHeaderForReads();
+        final String inputFileName = getReadSourceName();
+
         // Class where real metric-collection work is delegated to.
         final InsertSizeMetricsCollectorSpark collector = new InsertSizeMetricsCollectorSpark(filteredReads,
-                                                                                              // METRIC_ACCUMULATION_LEVEL,
-                                                                                              getHeaderForReads().getReadGroups(),
+                                                                                              readsHeader,
+                                                                                              METRIC_ACCUMULATION_LEVEL,
                                                                                               DEVIATIONS_TOL);
 
-        final SAMFileHeader readsHeader = getHeaderForReads();
-        final String inputFileName = getReadSourceName();
-
+        // similar to Picard::CollectInsertSizeMetrics.finish()
         if(OUTPUT != null) {
-            writeMetricsFile(collector, readsHeader, inputFileName);
```

Comment:
I am not quire sure if I understand this comment. This if check is here so that if the cmd line argument specifies an file output, then it will be generated. Otherwise the metrics just "lives in memory" (it is calculated in `InsertSizeMetricsCollectorSpark.java`, which provides getters). So I think my confusion about how this metrics collector will be used is generating the issue here: is it going to provide some results that other tools can call? or simply to be run onetime like a cmd line tool?